toolkit/tools/internal/safechroot/safechroot.go

```diff
@@ -689,3 +700,66 @@ func (c *Chroot) GetMountPoints() []*MountPoint {
 	mountPoints := append([]*MountPoint(nil), c.mountPoints...)
 	return mountPoints
 }
+
+// stopGPGComponents stops gpg-agent and keyboxd if they are running inside the chroot.
+//
+// A GPG agent may have been started while the chroot was in use. Newer versions of "gnupg2" will also start keyboxd.
+// E.g. when installing the azurelinux-repos-shared package, a GPG import occurs. This starts the gpg-agent process inside the chroot.
+// To be able to cleanly exit the setup chroot, we must stop it.
+func (c *Chroot) stopGPGComponents() (err error) {
+	_, err = exec.LookPath("gpgconf")
+	if err != nil {
+		// gpgconf is not installed, so gpg-agent is not running.
+		logger.Log.Debugf("gpgconf is not installed, so gpg-agent is not running: %s", err)
+		return
+	}
+
+	err = c.UnsafeRun(func() (err error) {
+		// Split --list-components stdout into a list of name tags, one for each component
```

Comment:
This comment might now be better located above the definition of the `listGPGComponents` function, since that's closer to its implementation.